packages/ui-components/src/components/SelectList/SelectList.tsx

```diff
@@ -35,7 +35,11 @@ const ListWrapper = styled.div<{
 }>`
   overflow-y: auto;
   max-height: 100%;
-  ${({ $variant }) => ($variant === 'fullPage' ? TopBorder : FullBorder)};
+  ${({ $variant }) => {
+    if ($variant === 'fullPage') return FullBorder;
+    if ($variant === 'inline') return TopBorder;
+    return '';
+  }}
```

Comment:
GitHub won’t let me comment on the unchanged line, but line 34 is perhaps better as `Pick<SelectListProps, 'variant’>`

Separately, capitalising `FullBorder` and `TopBorder` makes them look like React components. Would this be a bit clearer?

```suggestion
  ${({ $variant = 'borderless' }) => {
    if ($variant === 'fullPage') return fullBorder;
    if ($variant === 'inline') return topBorder;
    return '';
  }}
```